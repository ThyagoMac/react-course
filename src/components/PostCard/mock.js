export const mockPost = {
  id: 1,
  cover: "img/img.png",
  title: "Title Example",
  body: "Body example",
}