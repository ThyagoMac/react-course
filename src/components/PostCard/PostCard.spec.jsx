import { render, screen } from "@testing-library/react";
import { PostCard } from "./index.jsx";

import { mockPost } from "./mock.js";

describe("<PostCard />", () => {
  it("Render postcard", () => {
    /* debugger test
    const { debug } = render(<PostCard post={mockPost} />);
    debug();
    */

    render(<PostCard post={mockPost} />);

    const postCard = screen.getByRole("img", { name: /title example/i });
    const heading = screen.getByRole("heading", { name: /title example/i });
    const p = screen.getByText(/body example/i);

    expect(postCard).toHaveAttribute("src", mockPost.cover);
    expect(heading).toBeInTheDocument();
    expect(p).toBeInTheDocument();
  });

  it("Should match snapshot", () => {
    const { container } = render(<PostCard post={mockPost} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});