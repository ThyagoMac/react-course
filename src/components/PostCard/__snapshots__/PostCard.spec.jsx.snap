// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Title Example"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      (
      1
      ) 
      Title Example
    </h2>
    <p>
      Body example
    </p>
  </div>
</div>
`;
